src/vivarium_gates_nutrition_optimization/components/pregnancy.py

```diff
@@ -38,36 +40,82 @@ def columns_created(self):
             "pregnancy_duration",
         ] + self.new_children.columns_created
 
-    def setup(self, builder: Builder):
-        super(BaseDiseaseState, self).setup(builder)
+    @property
+    def sub_components(self):
+        return super().sub_components + [self.new_children]
 
+    @property
+    def initialization_requirements(self) -> Dict[str, List[str]]:
+        return {
+            "requires_columns": [self.model],
+            "requires_values": ["birth_outcome_probabilities"],
+            "requires_streams": [],
+        }
+
+    def setup(self, builder: Builder):
+        """Performs this component's simulation setup.
+
+        Parameters
+        ----------
+        builder : `engine.Builder`
+            Interface to several simulation tools.
+        """
+        super(DiseaseState, self).setup(builder)
         self.clock = builder.time.clock()
 
-        view_columns = self.columns_created + [self._model, "alive"]
-        self.population_view = builder.population.get_view(view_columns)
-        builder.population.initializes_simulants(
-            self.on_initialize_simulants,
-            creates_columns=self.columns_created,
-            requires_columns=[self._model],
-            requires_values=["birth_outcome_probabilities"],
+        prevalence_data = self.load_prevalence_data(builder)
+        self.prevalence = builder.lookup.build_table(
+            prevalence_data, key_columns=["sex"], parameter_columns=["age", "year"]
         )
-        self.prevalence = self.get_prevalence_table(builder)
-        self.birth_prevalence = self.get_birth_prevalence_table(builder)
+
+        birth_prevalence_data = self.load_birth_prevalence_data(builder)
```

Comment:
What's the reason for all these changes?

Reply:
I essentially abandoned my previous refactor of DiseaseState and made pregnancy state conform to the new refactored DiseaseState from VPH, and the format is slightly different in things like `foo = load_foo` instead of `foo = get_foo` and whether certain things get their own functions in setup() or are just done directly within setup (cf the mortality pipeline modifier)